Add support for IFLA_MIN_MTU and IFLA_MAX_MTU in LinkAttrs

AlekseyMamedoff opened the pull request:
Added attributes 	MinMTU, MaxMTU to LinkAttrs struct
Added cases for IFLA_MIN_MTU, IFLA_MAX_MTU
Signed-off-by: Aleksey Mamedov <alekseymamedoff@gmail.com>

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->

## Summary by CodeRabbit

* **New Features**
  * Network links now expose minimum and maximum transmission unit (MTU) bounds, enabling applications to query supported MTU ranges for network interfaces.

* **Tests**
  * Added comprehensive test coverage for MTU limit parsing and deserialization.

<!-- review_stack_entry_start -->

[![Review Change Stack](https://storage.googleapis.com/coderabbit_public_assets/review-stack-in-coderabbit-ui.svg)](https://app.coderabbit.ai/change-stack/vishvananda/netlink/pull/1102?utm_source=github_walkthrough&utm_medium=github&utm_campaign=change_stack)

<!-- review_stack_entry_end -->

<!-- end of auto-generated comment: release notes by coderabbit.ai -->

Commit:
Add support for IFLA_MIN_MTU and IFLA_MAX_MTU in LinkAttrs
Signed-off-by: Aleksey Mamedov <alekseymamedoff@gmail.com>

lazysegtree commented:
@AlekseyMamedoff , Fix for `TestXfrmStateWithPcpunumWithoutSADir`, the test which is failing in your PR, was recently merged in main. Please rebase with main.

AlekseyMamedoff commented:
@lazysegtree Done. Test passed. Thx.

lazysegtree commented on link_linux.go:
```diff
@@ -2248,6 +2248,10 @@ func LinkDeserialize(hdr *unix.NlMsghdr, m []byte) (Link, error) {
 			base.Name = string(attr.Value[:len(attr.Value)-1])
 		case unix.IFLA_MTU:
 			base.MTU = int(native.Uint32(attr.Value[0:4]))
+		case unix.IFLA_MIN_MTU:
```
Also a unit test would be great something like

```go
func TestLinkMinMaxMTU(t *testing.T) {
	tearDown := setUpNetlinkTest(t)
	defer tearDown()

	iface := &Dummy{LinkAttrs{
		Name: "foo",
	}}
	if err := LinkAdd(iface); err != nil {
		t.Fatal(err)
	}

	link, err := LinkByName("foo")
	err = LinkSetMinMTU(link, 1000)
	if link.Attrs().MinMTU != 1000 {
		t.Fatal("MinMTU not correct")
	}
}
```

EmilyShepherd replied:
I agree. Adding Setters for MaxMTU and MinMTU sizes does not make sense because they can't be set from userspace. They are a read-only view of what the interface is telling the kernel its minimum or maximum accepted MTU is.

AlekseyMamedoff replied:
Since the fields are read-only, we only need to check that they are parsed correctly into the structure LinkAttrs. I've added a corresponding test. Is it enough?

vanytsvetkov replied:
Hm, I think the following approach would be more observable:

1. Create a dummy interface with the kernel-defined `min_mtu` and `max_mtu` values:

   `ip link add name dummy0 type dummy`

2. Get the interface by name:

   `ip -d link show dummy0`

3. Parse the `min_mtu` and `max_mtu` values from the output:

```
67: dummy0: <BROADCAST,NOARP> mtu 1500 qdisc noop state DOWN mode DEFAULT group default qlen 1000
    link/ether b6:38:bd:4d:b0:95 brd ff:ff:ff:ff:ff:ff promiscuity 0 allmulti 0 minmtu 0 maxmtu 0 
    dummy addrgenmode eui64 numtxqueues 1 numrxqueues 1 gso_max_size 65536 gso_max_segs 65535 tso_max_size 65536 tso_max_segs 65535 gro_max_size 65536 gso_ipv4_max_size 65536 gro_ipv4_max_size 65536
```


The kernel sets `min_mtu` and `max_mtu` itself, as can be seen in the Linux source code in [dummy.c](https://github.com/torvalds/linux/blob/5200f5f493f79f14bbdc349e402a40dfb32f23c8/drivers/net/dummy.c#L125-L126):

```c
dev->min_mtu = 0;
dev->max_mtu = 0;
```

So we can expect the corresponding fields and values to be present in the netlink message.

EmilyShepherd replied:
> Hm, I think the following approach would be more observable:

I think we are at risk of overthinking this at this point.

The PR has added tests and those tests meaningfully cover the functionality the PR has added.

Given that this functionality is useful, and is not complex at all, it would be wonderful to get merged in, especially as the PR has been open for almost a year now.

AlekseyMamedoff replied:
@aboch ping

Commit:
Adds test coverage for parsing of `IFLA_MIN_MTU` and `IFLA_MAX_MTU`
attributes in `LinkDeserialize`.

The test validates that MTU limits are correctly deserialized
from rtnetlink attributes into `LinkAttrs`.
